### DISCUSSION TITLE
Wled app can be used for the mesmerizer

### DISCUSSION THREAD

**Discussion by heidepiek** [PlummersSoftwareLLC/NightDriverStrip#529] [Show and tell]
By chance I discovered that when I opened the wled app the app showed me the ip address of the mesmerizer. I think that  ip address was used once for the wled of 1 of my projects. And that the router has now given that ip address to mesmerizer because that address was no longer used by another device. Out of curiosity clicked the link, and look and behold the mesmerizer ui came up. Also, is it an idea to give the mesmerizer a standby mode? The off button on the wled app works on the wled devices, but not on the mesmerizer. 

https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/90329561-65bd-4836-b997-cdd42d12c0e4

Harrie

**Comment by robertlipe** [comment]
The remote interface already has an 'off' feature. I think it leaves the
actual effect running and just stops the display, but I'm not sure. Is that
what you're looking for?

I'm not near my mesmerizer right now, but there's a power icon in the left
panel. It's shown in your video on the bottom. Does that get you what you
want?

I don't think it goes into full CPU low-power mode. Too much stuff gets
weird when you do that and putting the 300mA CPU into low-power mode is a
bit silly when the LEDs take 6A or more.

**Comment by rbergen** [comment]
As far as WLED is concerned, the app just seems to load the on-board web UI on Mesmerizer, under the WLED header. From the perspective of NightDriverStrip, no effort was made - or is planned - to make that integration work. Which is also one reason why pressing the power button in the WLED app won't do anything - whatever request that sends to Mesmerizer will fall on deaf ears at the Mesmerizer side.

To be frank I don't know what a "stand-by mode" for NightDriverStrip would look like. As in: what would the device stop doing on the one hand and still do on the other when it's on stand-by?

The "off" button on the IR remote just decreases the brightness at the moment. One could use that to almost turn off the visuals; the mimimum brightness is 10 out of 255.

The power menu button in the web UI doesn't power anything off, but shows a number of options to (reset-and-)reboot.

**Comment by davepl** [comment]
As you note, the OFF actually dims the brightness by one step, it doesn’t turn the effect off.  I don’t see a need to pause or turn it off, but that doesn’t meant there isn’t one, just that I can’t think of one!

 _Dave

**Reply by heidepiek** [reply]
Was aware that the off and on buttons were used to control the brightness of the screen. Off to dim. And on to bring the screen back to full brightness. But what I meant by standby is, use an extra gpio pen to control a relay to turn the matrix off and on again. When I leave my mancave turn the mesmerizer (and I mean the screen because it takes the amps) off and leave the esp on. So when a enter the room the next day i can turn it on again. I know that you can pull the plug out off the wall and it wil all go out. But that was not my intention. T`was just an idea. My wled projects do get switched that way. Hence the power button. A specially made pcb with a relay that interrupts the voltage to the LEDs when you turn off the wled. But the esp does stay online. So when you enter the room again you can quickly turn the leds back on without waiting for the esp to restart and log in to the router. 

Harrie

**Comment by kriztioan** [comment]
A global ON/OFF for the effects, preferably via a WebRequest endpoint, would benefit me as I can then incorporate it into my home automation, avoiding the need for an additional smart socket. I did manage to use the `disableEffect` endpoint to disable the single effect I'm running on a schedule, but unfortunately the `enableEffect` does not re-enable it again.

```shell
curl --data "effectIndex=0" nightdriverstrip.centralpark.lan/disableEffect
```

TLDR; +1 for an ON/OFF feature.

**Comment by davepl** [comment]
As long as any remote key restores it, I guess I could see it being useful!

**Reply by heidepiek** [reply]
The esp won`t restart, it is only the relay powering the matrix that is off. Just use one key to power toggle the gpio specified for the relay. That is also how wled does it. But how, that i leave to the programmers.😌

**Comment by rbergen** [comment]
It sort of seems to me we now have two requests, one hardware-driven specific for Mesmerizer (from @heidepiek) and one more general and more software-oriented (from @kriztioan). As we're talking about GPIO pins and remote control buttons on an IR remote that I believe is pretty solidly booked, I'll leave that challenge to someone else.

What I also see in @kriztioan's suggestion is a bug report that enabling a single disabled effect doesn't work. That's something I **will** put on my list of things to look into.

**Reply by heidepiek** [reply]
You could use the power button in the web ui. Not only for reboot, reset device configuration and reset effect settings. But also for turning mesmerizer off. And mesmerizer off i mean a gpio pin that turns of the relay. Doesn't have to be special with the remote control! But then you can put the mesmerizer on standby.
Maybe someone is up to the challenge.  From what I've seen on here, what these geniuses can do, it should be a piece of cake to control 1 gpio pen with a button in the web ui. 😁

**Comment by robertlipe** [comment]
Do you need new ui for this? If brightness < 20% release FET. Off just
becomes a side effect of being dim.. existing code would them turn
brightness back on which would pop the fet again.

This seems awfully specialized, though, and I'm not sure it belongs in a
general purpose build.. we already have quite enough option avalanche. But
we'd still have to make the pin configurable, hopefully without further
polluting global namespace.

Let me think on it if the idea of tying it to bright ess. Works in general
for you.

**Reply by heidepiek** [reply]
It was just an idea, doesn't have to be difficult in my opinion. Nothing needs to be changed in the rest of the code I think. Just 1 gpio pen controlled by a command in the web ui. Nothing else happens to the esp and what it does. You only switch the gpio pen. And on the gpio pen the relay which controls the matrix. The esp just keeps doing what it does. Don't know how else to put it. It seems a piece of cake to me for the programmers to just assign 1 gpio pen that you make high or low with a button in the web ui. It need not be linked to the brightness of the matrix. You just turn the matrix off with the gpio pin on the relay that powers the matrix. But maybe I am thinking too easily! At wled's pcb they also just designated 1 gpio pin that operates a relay to turn off the power supply when they make the gpio pin low. and turn it back on by making it high. Nothing more, nothing less. The esp with the firmware on it just keeps running. The advantage is when you instruct the esp to turn the relay back on is that the esp does not have to restart and log on to the router again. 


https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/104028471/4e4ac3c1-a271-41d8-b160-08ea1ea22c0b

**Comment by robertlipe** [comment]
On Mon, Nov 20, 2023, 1:49 PM Harke Bosgraaf ***@***.***>
wrote:

> It was just an idea, doesn't have to be difficult in my opinion. Nothing
> needs to be changed in the rest of the code I think.
>

Thinking more, I'm happier with this being a web setting to control which
pin than another compile-time configuration.  That overcomes one of my
objections to configuration bloat.

>  the gpio pen the relay which
>
I'm not going to design the electronics here, but you'll need a SSR or at
least a FET/transistor to handle the current of a relay and a diode to
suppress the spike when the field collapses. The gpios on these things just
aren't meant for that kind of current.

 It seems a piece of cake to me for the programmers to just assign 1 gpio
> pen
>

Tip: when asking others to do work for you, don't estimate how easy it is.
😉

Wiggling the signal is easy. Making a sensible UI for using it and
configuring it is harder. It's admittedly not rocket science, but doing
things well is rarely super trivial.

To me, even starting the web interface seems like enough of a distraction
that I'd personally never click that button. That's why we're kicking
around how to tie it to existing remote interface.

>  running. The advantage is when you instruct the esp to turn the relay
> back on is that the esp does not have to restart and log on to the router
> again.
>

Makes sense. As described, even running both cores and the radio, the esp
pulls nothing compared to the panels or strips.

>
>

**Reply by heidepiek** [reply]
Sorry robert, I look up to those men who conceive and make all this. I also know that it is difficult stuff to program all this. If I could do it, I would help. And indeed there has to be a piece of electronics for the relay to work with the output of the gpio pin (transistor). That is also incorporated on the wled board. But it is a good intention. It seems easy to turn the mesmerizer on and off this way. Don't have to take the power off every time, and the esp doesn't have to start up and log in every time. I certainly don't mean for these guys to say it's easy, just have a lot of faith that they can do it.  Even if other people don't use it. I was thinking just 1 gpio pen that you can click in the web ui that it gets high at 1 press and low one more press. Nothing more, nothing less. Maybe someone will pick that up sometime, sorry if I seem a bit pushy. Do not mean it!
